### PR TITLE
`seqcli index (list|suppress)` and `seqcli expressionindex (create|list|remove)` commands

### PR DESCRIPTION
**Pull request by liammclennan**

### CODE REVIEW

File: src/SeqCli/Templates/Export/EntityName.cs [hunk 1]
```diff
@@ -15,6 +15,11 @@ public static string FromEntityType(Type entityType)
 
     public static string ToResourceGroup(string resource)
     {
+        if (resource.Equals("expressionindex"))
+        {
+            return "expressionindexes";
+        }
```
Comment (nblumhardt) [inline src/SeqCli/Templates/Export/EntityName.cs:21]: FWIW checking for trailing `x` and adding `es` would work in most cases, matching the `y` case below?

File: src/SeqCli/Cli/Commands/ExpressionIndex/CreateCommand.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    readonly OutputFormatFeature _output;
+
+    string? _expression;
+
+    public CreateCommand(SeqConnectionFactory connectionFactory, SeqCliConfig config)
+    {
+        _connectionFactory = connectionFactory ?? throw new ArgumentNullException(nameof(connectionFactory));
+
+        Options.Add(
+            "expression=",
```
Comment (nblumhardt) [inline src/SeqCli/Cli/Commands/ExpressionIndex/CreateCommand.cs:44]: `e=|expression=` ?

File: src/SeqCli/Cli/Commands/Index/ListCommand.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    {
+        var connection = _connectionFactory.Connect(_connection);
+
+        var list = _id is not null 
+            ? [await connection.Indexes.FindAsync(_id)]
+            : await connection.Indexes.ListAsync();
+        
+        _output.ListEntities(list);
+        
+        await Task.Delay(1);
```
Comment (nblumhardt) [inline src/SeqCli/Cli/Commands/Index/ListCommand.cs:59]: Is this needed?

File: src/SeqCli/Cli/Commands/Index/SuppressCommand.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    string? _id;
+
+    public SuppressCommand(SeqConnectionFactory connectionFactory, SeqCliConfig config)
+    {
+        if (config == null) throw new ArgumentNullException(nameof(config));
+        _connectionFactory = connectionFactory ?? throw new ArgumentNullException(nameof(connectionFactory));
+
+        Options.Add(
+            "i=|id=",
+            "The id of an index of a signal to suppress",
```
Comment (nblumhardt) [inline src/SeqCli/Cli/Commands/Index/SuppressCommand.cs:39]: Could drop `of a signal` here? The server might change around this point

File: src/SeqCli/Cli/Commands/Index/SuppressCommand.cs [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        if (_id == null)
+        {
+            Log.Error("An `id` must be specified");
+            return 1;
+        }
+
+        var connection = _connectionFactory.Connect(_connection);
+        var toSuppress = await connection.Indexes.FindAsync(_id);
+        if (toSuppress.IndexedEntityType != IndexedEntityType.Signal)
```
Comment (nblumhardt) [inline src/SeqCli/Cli/Commands/Index/SuppressCommand.cs:55]: Could we make the request and let the server give the reason for any 400 response?

File: src/SeqCli/Templates/Export/TemplateSetExporter.cs [hunk 6]
```diff
@@ -78,6 +79,13 @@ public async Task ExportTemplateSet()
             () => _connection.RetentionPolicies.ListAsync(),
             retentionPolicy => retentionPolicy.Id.Replace("retentionpolicy-", ""),
             templateValueMap);
+        
+        await ExportTemplates<ExpressionIndexEntity>(
+            id => _connection.ExpressionIndexes.FindAsync(id),
+            () => _connection.ExpressionIndexes.ListAsync(),
+            expressionIndex => expressionIndex.Id.Replace("expressionindex-", ""),
```
Comment (nblumhardt) [inline src/SeqCli/Templates/Export/TemplateSetExporter.cs:86]: `=> expressionIndex.Expression.All(char.IsLetterOrDigit) ? expressionIndex.Expression : expressionIndex.Id.Replace("expressionindex-", "")` might improve file naming for trivial indexes?

**Review by nblumhardt — COMMENTED** [review]
Had a few ideas/suggestions but LGTM :+1:

File: src/SeqCli/Cli/Commands/Index/ListCommand.cs (hunk 3 above)
Comment (liammclennan) [reply src/SeqCli/Cli/Commands/Index/ListCommand.cs:59]: nooooo

**Pull request merged**